sqlalchemy_ibmi/base.py

```diff
@@ -636,19 +637,41 @@ def get_rowcount(self):
     def fire_sequence(self, seq, type_):
         return self._execute_scalar(
             "SELECT NEXTVAL FOR " +
-            self.connection.dialect.identifier_preparer.format_sequence(seq) +
+            self.connection.dialect.preparer.format_sequence(seq) +
             " FROM SYSIBM.SYSDUMMY1",
             type_)
 
 
+class CoerceUnicode(sa_types.TypeDecorator):
+    """Coerce unicode to string"""
+
+    # TODO These methods are overridden from the default dialect and should be
+    #  implemented
+
+    def process_literal_param(self, value, dialect):
+        pass
+
+    def process_result_value(self, value, dialect):
+        pass
+
+    @property
+    def python_type(self):
+        pass
+
+    impl = sa_types.Unicode
+
+    def process_bind_param(self, value, dialect):
+        return value
+
+
 class IBMiDb2Dialect(default.DefaultDialect, PyODBCConnector):
 
     name = 'sqlalchemy_ibmi'
     max_identifier_length = 128
     encoding = 'utf-8'
     default_paramstyle = 'qmark'
     COLSPECS = COLSPECS
```

Comment:
This also looks like it shouldn't have been upper-cased in the class: https://docs.sqlalchemy.org/en/13/core/internals.html

> colspecs
>    A dictionary of TypeEngine classes from sqlalchemy.types mapped to subclasses that are specific to the dialect class. This dictionary is class-level only and is not accessed from the dialect instance itself.

Reply:
I opened https://github.com/IBM/sqlalchemy-ibmi/issues/41 to tackle that.